syntax="proto3";

package greet;

service Greeter {
	rpc createUser (User) returns (Users) {}
	rpc getUser (requestUser) returns (Users) {}
	rpc deleteUser (User) returns (Users) {}
	rpc updateUser (requestUpdate) returns (Users) {}
	rpc mensagemVazia (vazio) returns (vazio) {}
}

message mensagemVazia {}

message User {
	int64 id = 1;
	string valores = 2;
}

message Users {
	int64 id = 1;
	string valores = 2;
}

message requestUpdate {
	int64 id = 1;
	string valores = 2;
}

message requestUser {
	int64 id = 1;
}